components/dashboard/src/settings/Notifications.tsx

```diff
@@ -13,47 +13,63 @@ import settingsMenu from "./settings-menu";
 
 export default function Notifications() {
     const { user, setUser } = useContext(UserContext);
-    const isTransactionalMail = !user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails;
-    const toggleTransactionalMail = async ()=>{
+
+    const [isTransactionalMail, _setTransactionMail] = useState(!user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails);
+    const isTransactionalMailRef = useRef(isTransactionalMail);
+    const setTransactionMail = (data: boolean) => {
+        isTransactionalMailRef.current = data;
+        _setTransactionMail(data);
+    };
+
+    const toggleTransactionalMail = async () => {
         if (user) {
             user.additionalData = {
                 ... {
-                    ... user.additionalData,
+                    ...user.additionalData,
                     emailNotificationSettings: {
-                        ... user.additionalData?.emailNotificationSettings,
-                        disallowTransactionalEmails: !!isTransactionalMail
```

Comment:
why removing? I thought this would ensure `boolean` type

Reply:
`isTransactionalMail` is actually boolean. There is no need for `!!`, is there?